tests/robot-tests/tests/visual_testing/tables_and_charts.robot

```diff
@@ -0,0 +1,163 @@
+        user takes html snapshot of element    ${data_block}
+        ...    ${SNAPSHOT_FOLDER}/${content_block.release_id}/${CONTENT_SECTIONS_FOLDER}/${content_block.content_block_id}-${content_block.chart_type}-chart.html
+    END
+    user closes accordion section with id    content-${content_block.content_section_position}    id:content
+    log content block details    ${content_block}    Content Block    ${table_filepath}    ${chart_filepath}
+
+Check Secondary Stats Table
+    [Arguments]    ${content_block}
+    user scrolls to element    id:content
+    user waits until page contains element    id:releaseHeadlines-tables-tab
+    user scrolls to element    id:releaseHeadlines-tables-tab
+    user clicks element    id:releaseHeadlines-tables-tab
+    user waits until page contains element    testid:dataTableCaption
+    ${filepath}=    user takes screenshot of element    id:releaseHeadlines-tables
+    ...    ${SNAPSHOT_FOLDER}/${content_block.release_id}/${SECONDARY_STATS_FOLDER}/${content_block.content_block_id}-table.png
+    user takes html snapshot of element    id:releaseHeadlines-tables
+    ...    ${SNAPSHOT_FOLDER}/${content_block.release_id}/${SECONDARY_STATS_FOLDER}/${content_block.content_block_id}-table.html
+    log content block details    ${content_block}    Secondary Stats Table    ${filepath}
+
+Check Key Stats Table
+    [Arguments]    ${content_block}
+    user waits until page contains element
+    ...    xpath://div[@data-testid="keyStat"][${content_block.content_block_position}]
+    user waits until parent contains element
+    ...    xpath://div[@data-testid="keyStat"][${content_block.content_block_position}]    testid:keyStat-value
+    ${filepath}=    user takes screenshot of element
+    ...    xpath://div[@data-testid="keyStat"][${content_block.content_block_position}]
+    ...    ${SNAPSHOT_FOLDER}/${content_block.release_id}/${KEY_STATS_FOLDER}/${content_block.content_block_id}-table.png
+    user takes html snapshot of element    xpath://div[@data-testid="keyStat"][${content_block.content_block_position}]
+    ...    ${SNAPSHOT_FOLDER}/${content_block.release_id}/${KEY_STATS_FOLDER}/${content_block.content_block_id}-table.html
+    log content block details    ${content_block}    Key Stats Table    ${filepath}
+
+Log Content Block Details
+    [Arguments]
+    ...    ${content_block}
+    ...    ${type_description}
+    ...    ${table_snapshot_filepath}
+    ...    ${chart_snapshot_filepath}=
+    Log to console    \t\tTable snapshot taken for block ${content_block.content_block_id}
+    IF    "${chart_snapshot_filepath}" != ""
+        Log to console    \t\tChart snapshot taken for block ${content_block.content_block_id}
+    END
+
+user waits for chart to appear
+    [Arguments]    ${chart_type}    ${data_block}
+    IF    "${chart_type}" == "map"
+        user waits until element contains map chart    ${data_block}
+        sleep    1
```

Comment:
Why is there a sleep here?

Reply:
It's to let Leaflet settle in on a zoom level and pull in and apply additional styling - agree it's not great but was ending up with various tiny visual differences and sometimes a massively zoomed in map without!

Happy to try to improve it but I don't know when we'll get time to just yet.